File: product-consumer-service/src/main/java/de/hdi/product_consumer_service/model/Order.java
package de.hdi.product_consumer_service.model;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
public class Order {
    private Integer id;
    private String customerName;
    private String customerEmail;
    private String status;
    private java.util.List<OrderItem> items;
    private Double totalAmount;
}

File: order-kotlin-service/src/main/kotlin/de/hdi/order_kotlin_service/model/Order.kt
package de.hdi.order_kotlin_service.model

import java.time.LocalDateTime
import java.util.UUID

data class Order(
    val id: String = UUID.randomUUID().toString(),
    val productId: Int,
    val quantity: Int,
    val totalPrice: Double,
    val customerEmail: String,
    val createdAt: LocalDateTime = LocalDateTime.now(),
    val status: OrderStatus = OrderStatus.PENDING
)

enum class OrderStatus {
    PENDING, CONFIRMED // etc.
}

data class OrderRequest(
    val productId: Int,
    val quantity: Int,
    val customerEmail: String
)

data class OrderResponse(
    val id: String,
    val productId: Int,
    val productName: String,
    val quantity: Int,
    val totalPrice: Double,
    val customerEmail: String,
    val createdAt: LocalDateTime,
    val status: OrderStatus
)